docs/datastream/datastream.md

```diff
@@ -0,0 +1,193 @@
+# DATASTREAM FORMAT
+
+The format of the data in the data stream is defined using protobuf. Latest version can always be found here. This document describes them as they are in the date of creation of the document.
+
+## Entry types
+
+### BookMark
+
+​​​​- Type
+
+​​​​- Value
+
+**Notes**
+There are bookmarks for Batches (type = 1) and L2Blocks (type = 2).
+
+### BatchStart
+
+​​​​- Number
+
+​​​​- Type
+
+​​​​- ForkID
+
+​​​​- ChainID
+
+​​​​- Debug
+
+**Notes**
+
+* ForkID is 0 for genesis batch
+
+* Type can be:
+
+   - Regular: For normal batches
+
+   - Forced: For forced batches
+
+   - Injected: Batch 1 of a network with ForkID >= 7. It is generated by the smart contract during network creation.
+
+
+### L2Block
+
+​​​​- Number
+
+​​​​- BatchNumber
+
+​​​​- Timestamp
+
+​​​​- DeltaTimestamp
+
+​​​​- MinTimestamp
+
+​​​​- L1BlockHash
+
+​​​​- L1InfoTreeIndex
+
+​​​​- Hash (BlockHash)
+
+​​​​- StateRoot
+
+​​​​- GlobalExitRoot
+
+​​​​- Coinbase
+
+​​​​- BlockGasLimit
+
+​​​​- BlockInfoRoot
+
+- StateRoot
+
+​​​​- Debug
+
+**Notes**
+
+* **Hash** - hash calculated by the node and used in the RPC
+
+* **StateRoot** 
+
+  - For ForkIDs < 7: Final state root of the L2Block (see also *IntermediateStateRoot* in *L1Transacton*)
+
+  - For ForkIDs >=7: Final state root of the L2Block
+
+* **L1BlockHash**, only For ForkIDs >= 7
+  
+     - For regular batches: ParentHash of the L1 block when the GER was inserted in the L1InfoTree
+
+     - If batch is Forced or Injected: BlockHash of the parent L1 block when the forced or injected happen in L1
+
+* **DeltaTimestamp**: difference between current and previous block timestamps
+
+* **MinTimestamp**: L1InfoTree contains: GER, L1BlockHash and minTimeStamp
+
+  - For ForkID >= 7: Timestamp when a GER was inserted in the L1InfoTree.
+  
+  - For ForkID < 7: Block Timestamp
+
+  - For Genesis block: 0
+
+* **BlockGasLimit**: Currently unused (set to 0)
+
+* **BlockInfoRoot**: Only for ForkIDs >= 7, 0x0000… for ForKIDs < 7
+
+### Transaction
+
+​​​​- L2BlockNumber
+
+​​​​- Index
+
+​​​​- IsValid
+
+​​​​- Encoded
+
+​​​​- EffectiveGasPricePercentage
+
+​​​​- IntermediateStateRoot
+
+​​​​- Debug
+
+**Notes**
+
+* EffectiveGasPricePercentage: Only available for ForkIDs >= 5, 0 in other ForkIDs.
+
+* IntermediateStateRoot: For ForkIDs < 7 is the resulting SR_A of the TX execution, previous to storing it in the system SC and so getting a new SR_B.
```

Comment:
Do we need to specify this for just pre etrog?  We can keep it in place for all forks to keep things simpler maybe

Reply:
Then again if we do 0x0 it for post etrog it is less work for the sequencer as it will never need to store these values